ui/app/templates/rules.html

```diff
@@ -69,8 +69,9 @@ <h3 class="panel-title">
         <span ng-show="!rule_id && rule.scheduled_change === 'delete'" class="label label-danger">Pending Delete</span>
         <button ng-show="!rule_id && rule.scheduled_change === null" class="btn btn-warning btn-xs" ng-click="openNewScheduledRuleChangeModal(rule)">Schedule an Update</button>
         <button ng-show="!rule_id && rule.scheduled_change === null" class="btn btn-danger btn-xs" ng-click="openNewScheduledDeleteModal(rule)">Schedule for Deletion</button>
-        <button ng-show="!rule_id" class="btn btn-default btn-xs" ng-click="openUpdateModal(rule)">Update</button>
-        <button ng-show="!rule_id" class="btn btn-default btn-xs" ng-click="openDeleteModal(rule)">Delete</button>
+        <button ng-show="ruleSignoffsRequired(rule).length" class="btn btn-warning btn-xs">Signoffs required</button>
+        <button ng-show="!rule_id" class="btn btn-default btn-xs {{ruleSignoffsRequired(rule).length && 'disabled'}}" ng-click="openUpdateModal(rule)">Update</button>
+        <button ng-show="!rule_id" class="btn btn-default btn-xs {{ruleSignoffsRequired(rule).length && 'disabled'}}" ng-click="openDeleteModal(rule)">Delete</button>
```

Comment:
I'm also not entirely sure what to do here. A few thoughts:
* If we're disabling the buttons, maybe we can add title text that says "Cannot modify this object directly due to signoff requirements" (or something shorter to the same effect)
* In #394, I started hiding buttons entirely when they weren't useful. That could be an option too. Do you feel strongly one way or the other? We should make sure we do the same thing.
* Either way, I don't *think* we need the "Signoffs required" item -- having the only functional buttons be "Schedule..." should be enough to make that clear.

Reply:
I don't feel like there's a difference between buttons being present but disabled without explanation and buttons being completely absent. I feel like having a disabled button with information explaining why that button is disabled is the best UI. I'm worried less about the release managers who probably know why it's disabled than for the next developer who has to try to figure out what the interface is doing and why. I'm worried less about the release managers who probably know why it's disabled than for the next developer who has to try to figure out what the interface is doing and why.

I like the idea of "title text", but I can't figure out a way to implement it in Angular. Button `title` attributes aren't displayed on hover by default. Bootstrap offers a "tooltip" mechanism but I can't get it to work easily. (It's implemented in Bootstrap JS, and it's opt-in, and it seems like the Bootstrap JS isn't included. Then, ideally, I'd have a clear-enough way to say "show this version of the button if it requires signoffs, and this version otherwise", but there's nothing like "else" in Angular as far as I can tell.)

I guess the right thing to do might be to hide the buttons and punt on tooltips till a later day?

Reply:
That sounds reasonable to me - we already do it for other things, too (eg: we hide "Schedule an Update" when a change is already scheduled for the Rule).